src/languages/snowflake/snowflake.formatter.ts

```diff
@@ -0,0 +1,282 @@
+import { expandPhrases } from 'src/expandPhrases';
+import Formatter from 'src/formatter/Formatter';
+import Tokenizer from 'src/lexer/Tokenizer';
+import { functions } from './snowflake.functions';
+import { keywords } from './snowflake.keywords';
+
+const reservedSelect = expandPhrases(['SELECT [ALL | DISTINCT]']);
+
+// https://docs.snowflake.com/en/sql-reference/sql-all.html
+//
+// 1. run in console on this page: $x('//tbody/tr/*[1]//a/span/text()').map(x => x.nodeValue)
+// 2. delete all lines that contain a sting like '(.*)', they are already covered in the list
+// 3. delete all lines that contain a sting like '<.*>', they are already covered in the list
+// 4. delete all lines that contain '…', they are part of a regex statement that can't be covered here
+// 5. Manually add 'COPY INTO'
+//
+// Steps 1-4 can be combined by the following script in the developer console:
+// $x('//tbody/tr/*[1]//a/span/text()').map(x => x.nodeValue) // Step 1
+//   filter(x => !x.match(/\(.*\)/) && !x.match(/…/) && !x.match(/<.*>/)) // Step 2-4
```

Comment:
This generated list is not really particularly useful. For example `INSERT` shouldn't be listed as a separate command, instead there should be `INSERT [OVERWRITE] INTO`.

I suggest you instead look at the `reservedCommands` in `sql.formatter.ts` and use it as a base and modify it to match Snowflake syntax. Also, it would be nice to keep things in the same order (e.g. starting with `WITH`, `FROM`, `WHERE`), so it's easier to compare with other other dialect definitions in *.formatter.ts files.

Reply:
Ok, I did this like in the `postgresql.formatter.ts`. I added the basic commands with the `{..} [...]` syntax, as well as the generated List because it contains many many useful phrases.